server/middleware/userValidation.js

```diff
@@ -0,0 +1,32 @@
+import Joi from 'joi';
+
+const validate = {
+  /**
+   * validate user submitted information
+   * prior to a successful user registration
+   * @param {object} req
+   * @param {object} res
+   * @param {request} next
+   */
+  validateRegistration(req, res, next) {
+    const userSchema = Joi.object().keys({
+      firstname: Joi.string().alphanum().min(4).max(15)
+        .required(),
+      lastname: Joi.string().alphanum().min(4).max(15)
+        .required(),
+      email: Joi.string().email({ minDomainAtomas: 2 })
```

Comment:
check here. It could be minDomainAtoms, not minDomainAtomas

Reply:
There is no mistake because everything works well!

Reply:
I'm gonna implement the second one instead. The one for retyping in the password!

Reply:
I will also implement the first one because of the clarification. You should clarify things for a better understanding!

Reply:
cool